src/main/java/zosjobs/DeleteJobs.java

```diff
@@ -0,0 +1,70 @@
+package zosjobs;
+
+import core.ZOSConnection;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import rest.*;
+import utility.Util;
+import utility.UtilIO;
+import zosjobs.input.DeleteJobParams;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DeleteJobs {
+
+    private static final Logger LOG = LogManager.getLogger(DeleteJobs.class);
+    private final ZOSConnection connection;
+
+    /**
+     * DeleteJobs Constructor
+     *
+     * @param connection connection information, see ZOSConnection object
+     * @author Nikunj Goyal
+     */
+    public DeleteJobs(ZOSConnection connection) {
+        this.connection = connection;
+    }
+
+    /**
+     * Delete a job that resides in a z/OS data set.
+     *
+     * @return job document with details about the submitted job
+     * @throws Exception error on submitting
+     * @author Frank Giordano
+     */
+    public Response deletejob( ) throws Exception {
+        return this.deleteJobCommon(null);
+    }
+
+    public Response deleteJobCommon(DeleteJobParams parms) throws Exception {
+        Util.checkNullParameter(parms == null, "parms is null");
+        Util.checkStateParameter(!parms.getJobId().isPresent(), "jobid not specified");
+        Util.checkStateParameter(!parms.getJobname().isPresent(), "jobname not specified");
+
```

Comment:
it seems modifyversion is required as you are using it below and as such we such check for it before using it with Util.checkStateParameter(parms.getModifyVersion().isEmpty(), "modify version not specified"